r/R/arrow-package.R

```diff
@@ -65,6 +65,14 @@
     # Disable multithreading on Windows
     # See https://issues.apache.org/jira/browse/ARROW-8379
     options(arrow.use_threads = FALSE)
+
+    # Try to set timezone database
+    if (requireNamespace("tzdb", quietly = TRUE)) {
+      tzdb::tzdb_initialize()
+      set_timezone_database(tzdb::tzdb_path("text"))
+    } else {
+      warning("tzdb not installed. Timezones will not be available.")
+    }
```

Comment:
@nealrichardson  @paleolimbot the R CMD CHECK doesn't seem to like us referencing `tzdb::` in `onLoad`. ([See this CI job](https://github.com/apache/arrow/runs/5400279568?check_suite_focus=true#step:6:1162).) Any thoughts on a better way to handle an optional dependency? (I think we only need this on Windows.)